lib/chef_zero/server.rb

```diff
@@ -314,7 +327,9 @@ def handle_socketless_request(request_env)
     #   true if the server is accepting requests, false otherwise
     #
     def running?
-      !@server.nil? && @running && @server.status == :Running
+      # Take a local copy of server in case it gets changed while we run
+      server = self.server
```

Comment:
And I would prefer that we don't shadow the `server` method with a `server` ivar - what about `current_server`?